common/include/pcl/pcl_macros.h

```diff
@@ -88,7 +88,7 @@
 #if defined(_MSC_VER)
   // nvcc on msvc can't work with [[deprecated]]
   #if !defined(__CUDACC__)
-    #define _PCL_DEPRECATED_IMPL(Message) [[deprecated(Message)]]
+    #define _PCL_DEPRECATED_IMPL(Message) __declspec(deprecated(Message))
```

Comment:
maybe provide a comment and url to the discussion on why this is required. A future developer will be unaware of your discoveries.

Reply:
This commit is actually outdated, it turns out that the `__declspec` approach will throw errors in some cases (see the Windows build errors for this commit). In the newest commit I'm not modifying the macros, but directly making the function inline on `MSVC` only. It seems to be compiling.